client/src/pages/queries/components/chart-tab.tsx

```diff
@@ -1,149 +1,105 @@
-import React, { cloneElement, useMemo, useState } from 'react'
-import { Empty, Form, Select, Input } from 'antd'
+import React, { useContext, useMemo, useState } from 'react'
+import { Button } from 'antd'
 import {
-  LineChartOutlined,
   AreaChartOutlined,
-  PieChartOutlined,
   BarChartOutlined,
+  LineChartOutlined,
+  PieChartOutlined,
+  PlusOutlined,
 } from '@ant-design/icons'
-import { QueryResult } from 'types/query'
-import { ChartTypes } from 'types/chart'
-import { chartGroups, chartList, chartOrder } from '../data/chart-list'
+import EditorContext from 'contexts/editor-context'
+import CreateChart from './create-chart'
+import ChartsList from './charts-list'
+import ChartDetail from './chart-detail'
 
-type ChartTabProps = {
-  data: QueryResult | null
-}
+export default function ChartTab() {
+  const { queryResult: data, isSaved } = useContext(EditorContext)
 
-export default function ChartTab({ data }: ChartTabProps) {
-  const [label, setLabel] = useState<string | undefined>()
-  const [values, setValues] = useState<string | string[] | undefined>()
-  const [title, setTitle] = useState<string | undefined>('Untitled Chart')
-  const [chartType, setChartType] = useState<ChartTypes>('line')
+  const [activeChartId, setActiveChartId] = useState<number>(-1)
```

Comment:
Instead of using state, we can use `useSearchParams` hook.

```tsx
const [searchParams, setSearchParams] = useSearchParams()
let activeChart = Number.parseInt(searchParams.get('chart'))
activeChart = isNaN(activeChart) ? searchParams.get('chart') : activeChart
```

Reply:
And for **new chart** page, set the `activeChart` value to `new-chart`.